scytale.yaml

```diff
@@ -1,49 +1,346 @@
+  # was chosen because it does not conflict with any of the other prometheus
+  # metrics or other machines in the xmidt cluster.  You may use any port you
+  # wish.
+  address: ":6303"
+
+  # metricsOptions provides the details needed to configure the prometheus
+  # metric data.  Metrics generally have the form:
+  #
+  # {namespace}_{subsystem}_{metric}
+  #
+  # so if you use the suggested value below, your metrics are prefixed like
+  # this:
+  #
+  # xmidt_scytale_{metric}
+  #
+  # (Optional)
+  metricsOptions:
+    # namespace is the namespace of the metrics provided
+    # (Optional)
+    namespace: "xmidt"
+    # subsystem is the subsystem of the metrics provided
+    # (Optional)
+    subsystem: "scytale"
+
+########################################
+#   Logging Related Configuration
+########################################
+
+# log configures the logging subsystem details
+log:
+  # file is the name of the most recent log file.  If set to "stdout" this
+  # will log to os.Stdout.
+  # (Optional) defaults to os.TempDir()
+  # file: "/var/log/scytale/scytale.log"
+  file: "stdout"
+
+  # level is the logging level to use - INFO, DEBUG, WARN, ERROR
+  # (Optional) defaults to ERROR
+  level: "DEBUG"
+
+  # maxsize is the maximum file size in MB
+  # (Optional) defaults to max 100MB
+  maxsize: 50
+
+  # maxage is the maximum number of days to retain old log files
+  # (Optional) defaults to ignore age limit (0)
+  maxage: 30
+
+  # maxbackups is the maximum number of old log files to retain
+  # (Optional) defaults to retain all (0)
+  maxbackups: 10
+
+  # json is a flag indicating whether JSON logging output should be used.
+  # (Optional) defaults to false
+  json: true
+
+########################################
+#   Fanout Related Configuration
+########################################
+
+# fanout describes the configuration used for faning out the http requests too.
+# defined https://godoc.org/github.com/xmidt-org/webpa-common/xhttp/fanout#Configuration
+fanout:
+  # endpoints are the URLs for each endpoint to fan out to.
+  # (Optional) if empty, endpoints are driven by service discovery
+  endpoints: ["https://petasos:6400/api/v2/device/send"]
+
+  # Authorization is the Basic Auth token.  There is no default for this field.
+  # (Optional) defaults to no auth token
+  # WARNING: This is an example auth token. DO NOT use this in production.
+  authorization: YXV0aEhlYWRlcg==
+
+  # transport is a way to overwrite the default golang http.Transport configuration.
+  # defined  https://golang.org/pkg/net/http/#Transport
+  # (Optional) defaults described below
+  transport:
+    # (Optional) defaults to 0, aka do not limit it
+    maxIdleConns: 0
+    # (Optional) defaults to 100
+    maxIdleConnsPerHost: 100
+    # (Optional) defaults to 0s, aka do not timeout while in idle
+    idleConnTimeout: "120s"
+
+  # fanoutTimeout is the timeout for the entire fanout operation.
```

Comment:
so this includes any waiting in scytale before starting the fanout (because of concurrency limits)? Something to make it more clear the difference between `fanoutTimeout` and `clientTimeout` would be great.